modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/table/index.tsx

```diff
@@ -12,66 +12,122 @@
  * details.
  */
 
-import {ClayButtonWithIcon} from '@clayui/button';
+import Button, {ClayButtonWithIcon} from '@clayui/button';
 import ClayTable from '@clayui/table';
 import classnames from 'classnames';
 
 import SettingsButton, {ActionObject} from '../settings-button';
 
-const {Body, Cell, Head, Row} = ClayTable;
-
-type TableProps = {
-	actions: ActionObject[];
-	data: {[keys: string]: string}[];
-	headers: TableHeaders[];
-	onClickRules?: (item: any, rowContent: any) => void;
-	setSortByDate?: () => void;
-	sortByDate?: string;
+type sort = {
+	[keys: string]: boolean;
 };
 
+type TableRowContentType = {[keys: string]: string};
+
 type TableHeaders = {
 	bold?: boolean;
 	centered?: boolean;
 	clickable?: boolean;
+	clickableSort?: boolean;
 	greyColor?: boolean;
 	hasSort?: boolean;
 	icon?: boolean;
 	key: string;
 	redColor?: boolean;
+	requestLabel: string;
 	type?: string;
 	value: string;
 };
 
+type TableProps = {
+	actions: ActionObject[];
+	data: {[keys: string]: string}[];
+	headers: TableHeaders[];
+	onClickRules?: (
+		item: TableHeaders,
+		rowContent: TableRowContentType
+	) => void;
+	onSaveCurrent?: (item: string) => void;
+	setSort?: (item: sort) => void;
+	setSortByOrder?: (item: string) => void;
+	sort?: sort;
+	sortByOrder?: string;
+	valuer?: string;
+};
+
+const {Body, Cell, Head, Row} = ClayTable;
+
 const Table: React.FC<TableProps> = ({
 	data,
 	headers,
 	actions,
-	sortByDate,
-	setSortByDate,
+	setSortByOrder,
+	setSort,
+	sort,
+	sortByOrder,
 	onClickRules = () => null,
+	onSaveCurrent,
 }) => {
+	const setCurrentHeaderPlus = (item: string) => {
+		onSaveCurrent?.(item);
+	};
+
+	const updateSort = (colunn: string) => {
+		const newSort: sort = {};
+
+		headers.forEach((item) => {
+			if (item.requestLabel === colunn) {
+				newSort[item.key] = true;
+
+				return;
+			}
+			newSort[item.key] = false;
+		});
+
+		setSort?.(newSort);
+	};
+
 	return (
 		<table className="border-0 ray-table show-quick-actions-on-hover table table-autofit table-list table-responsive">
 			<Head>
 				<Row className="ray-table-head">
-					{headers.map((header, index) => (
+					{headers.map((header, index: number) => (
```

Comment:
Could you put the type to header parameter?
```suggestion
					{headers.map((header: ???, index: number) => (
```

Reply:
Done!